crates/polars-ops/src/frame/pivot/mod.rs

```diff
@@ -192,112 +192,125 @@ fn pivot_impl(
 
     let mut final_cols = vec![];
 
+    let column_column_name = &columns[0];
+    let mut binding = pivot_df.clone();
+    let pivot_df: &DataFrame = if columns.len() > 1 {
+        let fields = columns
+            .iter()
+            .map(|s| pivot_df.column(s).unwrap().clone())
+            .collect::<Vec<_>>();
+        let columns_struct = StructChunked::new(column_column_name, &fields)
+            .unwrap()
+            .into_series();
+        binding.with_column(columns_struct).unwrap()
+    } else {
+        pivot_df
+    };
```

Comment:
in the base case, `pivot_df` isn't modified, the only call that happens to it is `pivot_df.clone`

does the `clone` here just increase the reference count, and so as `pivot_df` isn't modified, is zero-cost?

if so, then the base case (single-column) is unaffected by these changes